subsys/logging/log_backend_adsp.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include <sys/ring_buffer.h>
+#include <device.h>
+#include <assert.h>
+#include <soc.h>
+
+#define BUF_SIZE 64
+
+static struct ring_buf ringbuf;
+
+static u16_t magic = 0x55aa;
+static u16_t log_id;
+
+/*
+ * Log message format
+ * logging started with magic number 0x55aa followed by lo message id.
+ * Log message ended with null terminator and takes BUF_SIZE slot
+ */
+
+static void init(void)
+{
+	ring_buf_init(&ringbuf, CONFIG_LOG_BACKEND_ADSP_RINGBUF_SIZE,
```

Comment:
should be somehow ensured that `CONFIG_LOG_BACKEND_ADSP_RINGBUF_SIZE` is a multiple of `BUF_SIZE` otherwise when buffer is claimed it might not have `BUF_SIZE` available even when ring buffer has `BUF_SIZE` available.

Reply:
Fixed by adding a build assert